addons/website_slides/models/slide_channel.py

```diff
@@ -70,6 +72,13 @@ def _recompute_completion(self):
             completed_records._set_as_completed()
             completed_records._send_completed_mail()
 
+    @api.model_create_multi
+    def create(self, values_list):
+        for vals in values_list:
+            if vals.get('completed', False):
+                vals['completion_date'] = fields.Datetime.now()
```

Comment:
Why not a computed editable fields based on completed ?

Reply:
Indeed, it is simpler that way, thanks.